ttl: default to cluster setting select batch size in ttl job description

ecwall opened the pull request:
Fixes #111208

Previously the TTL job description diplayed LIMIT 0 if no select batch size
was specified for the table instead of properly diplaying the cluster setting
value.

This commit changes the code that generates the description to use the same
logic as the job itself which falls back to the cluster setting if no table
storage param is specified.

Release note: None

ecwall committed:
ttl: pass catpb.RowLevelTTL as pointer to setting accessor functions

The RowLevelTTL data is stored as a pointer in the table descriptor and can be
passed as one to getSelectBatchSize, getDeleteBatchSize, getDeleteRateLimit to
avoid copying the struct each call.

Release note: None

ecwall committed:
ttl: relocate TTL cluster settings

Move the TTL settings into the ttlbase package so they can be accessed by both
the ttljob and ttlschedule packages.

Release note: None

ecwall committed:
ttl: default to cluster setting select batch size in ttl job description

Fixes #111208

Previously the TTL job description diplayed LIMIT 0 if no select batch size
was specified for the table instead of properly diplaying the cluster setting
value.

This commit changes the code that generates the description to use the same
logic as the job itself which falls back to the cluster setting if no table
storage param is specified.

Release note: None

ecwall committed:
ttl: do not capture TTL job config

The TTL job config is captured, but unused so do not capture it.

Release note: None

ecwall committed:
ttl: move time forward inside th.executeSchedules

The time must be moved forward before TestingNudgeAdoptionQueue is called
so do it inside th.executeSchedules which is the only place
TestingNudgeAdoptionQueue is called.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/111227)
<!-- Reviewable:end -->

chrisseto commented on pkg/sql/ttl/ttlbase/ttl_helpers.go:
```diff
@@ -13,25 +13,90 @@ package ttlbase
+	defaultDeleteRateLimit = settings.RegisterIntSetting(
+		settings.TenantWritable,
+		"sql.ttl.default_delete_rate_limit",
+		"default delete rate limit for all TTL jobs. Use 0 to signify no rate limit.",
```
nit: can you specify the units on this or further describe how this number is interpreted?

edit: I see now this was just moved. Would be a nice to have to improve the documentation though :D

ecwall replied:
I'll change it now anyways to make it more clear.

chrisseto commented on pkg/sql/ttl/ttljob/ttljob.go:
```diff
@@ -40,35 +39,12 @@ import (
-		"whether the TTL job is enabled",
-		true,
-		settings.WithPublic)
+var jobEnabled = settings.RegisterBoolSetting(
```
Hmm, it feels a bit strange to not see all these settings grouped together. How would you feel about moving this to ttlbase as well? If that's not easy, maybe add a note indicating that settings for the job execution can be found in ttl base?

ecwall replied:
I'll move it as well for consistency.

chrisseto reviewed (approved):
LGTM! Nothing blocking 👍

ecwall committed:
ttl: clarify sql.ttl.default_delete_rate_limit description

Add rows per second units and add detail that it is per node per job.

Release note: None

ecwall commented:
bors r=chrisseto
